spec.bs

```diff
@@ -349,24 +347,38 @@ dictionary AuctionAdInterestGroupKey {
 The <dfn for=Navigator method>leaveAdInterestGroup(group)</dfn> method steps
 are:
 
-1. If [=this=]'s [=relevant global object=]'s [=associated Document=] is not [=allowed to use=] the
-  "[=join-ad-interest-group=]" [=policy-controlled feature=], then [=exception/throw=] a
-  "{{NotAllowedError}}" {{DOMException}}.
 1. Let |frameOrigin| be the [=relevant settings object=]'s [=environment settings object/origin=].
 1. [=Assert=] that |frameOrigin| is not an [=opaque origin=] and its [=origin/scheme=] is "`https`".
-1. Let |owner| be the result of [=parsing an origin=] with
-  |group|["{{AuctionAdInterestGroupKey/owner}}"].
-1. If |owner| is failure, [=exception/throw=] a {{TypeError}}.
-1. Let |name| be |group|["{{AuctionAdInterestGroupKey/name}}"].
 1. Let |p| be [=a new promise=].
-1. Run these steps [=in parallel=]:
-  1. Let |permission| be the result of [=checking interest group permissions=] with 
-    |owner|, |frameOrigin|, and false.
-  1. If |permission| is false, then [=queue a task=] to [=reject=] |p| with a
-     "{{NotAllowedError}}" {{DOMException}} and do not run the remaining steps.
-  1. [=Queue a task=] to [=resolve=] |p| with `undefined`.
-  1. [=list/Remove=] [=interest groups=] from the user agent's [=interest group set=] whose
-    [=interest group/owner=] is |owner| and [=interest group/name=] is |name|.
+1. If |group| [=map/is empty=]:
+  1. Let |instance| be [=this=]'s [=relevant global object=]'s [=Window/browsing context=]'s
+    [=browsing context/fenced frame config instance=].
+  1. If |instance| is null, [=exception/throw=] a {{TypeError}}.
+  1. Let |interestGroup| be |instance|'s [=fenced frame config instance/interest group descriptor=].
+  1. Run these steps [=in parallel=]:
+    1. [=Queue a task=] to [=resolve=] |p| with `undefined`.
+    1. If |interestGroup| is not null:
+      1. Let |owner| be |interestGroup|'s [=interest group descriptor/owner=].

```

Comment:
nit: would be slightly more clear if `owner` was called `ownerOrigin`, but shrug.

Reply:
Agreed but I think for consistency with the rest of spec we should leave it, otherwise I think it might be confusing why this one instance was different.  I low-key want bikeshed to do complete static-analysis type checking but that's a dream.

Reply:
> I low-key want bikeshed to do complete static-analysis type checking but that's a dream.

Sounds like a great 20% project (for someone else to do)